npm_and_yarn/lib/dependabot/npm_and_yarn.rb

```diff
@@ -97,88 +124,121 @@ def self.extract_var(error_message)
       },
       "YN0020" => {
         message: "Missing lockfile entry",
-        new_error: ->(_error, message) { Dependabot::DependencyFileNotFound.new(message) }
+        handler: ->(message, _error, _params) { Dependabot::DependencyFileNotFound.new(message) }
       },
       "YN0046" => {
         message: "Automerge failed to parse",
-        new_error: ->(_error, message) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
+        handler: ->(message, _error, _params) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
       },
       "YN0047" => {
         message: "Automerge immutable",
-        new_error: ->(_error, message) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
+        handler: ->(message, _error, _params) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
       },
       "YN0062" => {
         message: "Incompatible OS",
-        new_error: ->(_error, message) { Dependabot::DependabotError.new(message) }
+        handler: ->(message, _error, _params) { Dependabot::DependabotError.new(message) }
       },
       "YN0063" => {
         message: "Incompatible CPU",
-        new_error: ->(_error, message) { Dependabot::IncompatibleCPU.new(message) }
+        handler: ->(message, _error, _params) { Dependabot::IncompatibleCPU.new(message) }
       },
       "YN0071" => {
         message: "NM can't install external soft link",
-        new_error: ->(_error, message) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
+        handler: ->(message, _error, _params) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
       },
       "YN0072" => {
         message: "NM preserve symlinks required",
-        new_error: ->(_error, message) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
+        handler: ->(message, _error, _params) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
       },
       "YN0075" => {
         message: "Prolog instantiation error",
-        new_error: ->(_error, message) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
+        handler: ->(message, _error, _params) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
       },
       "YN0077" => {
         message: "Ghost architecture",
-        new_error: ->(_error, message) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
+        handler: ->(message, _error, _params) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
       },
       "YN0080" => {
         message: "Network disabled",
-        new_error: ->(_error, message) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
+        handler: ->(message, _error, _params) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
       },
       "YN0081" => {
         message: "Network unsafe HTTP",
-        new_error: ->(_error, message) { Dependabot::NetworkUnsafeHTTP.new(message) }
+        handler: ->(message, _error, _params) { Dependabot::NetworkUnsafeHTTP.new(message) }
```

Comment:
params is never used?

Reply:
Params is used here, https://github.com/dependabot/dependabot-core/pull/10257/files/a4486164a818184547958200f90a8fabbc1e8ae9#diff-69dbdae10606a6a47283c9645871927b0fb715a94146718638e77600f52bfb5cR215 and it can be used in future errors. So I need to keep it. Also in sorbet I couldn't pass typing if I don't add parameters. Only when you are not using parameter you can define it to start with `_`. Please let me know if you have suggestion. 

Just want to note that although now we are mostly not using it, anytime we can use it for new errors or improve the errors by getting something from the the given variables.